kolibri/plugins/learn/assets/src/views/ExploreLibrariesPage/index.vue

```diff
@@ -158,40 +182,39 @@
       });
 
       this.fetchDevices().then(devices => {
-        this.networkDevices = devices;
-        for (const device of this.networkDevices) {
+        const fetchDevicesChannels = devices.reduce((accumulator, device) => {
           const baseurl = device.base_url;
-          this.fetchChannels({ baseurl })
-            .then(channels => {
-              this.setNetworkDeviceChannels(device, channels.slice(0, 4), channels.length);
-            })
-            .catch(() => {
-              this.setNetworkDeviceChannels(device, [], 0);
-            });
-        }
+          accumulator.push(this.fetchChannels({ baseurl }));
+          return accumulator;
+        }, []);
+
+        Promise.allSettled(fetchDevicesChannels).then(devicesChannels => {
```

Comment:
There's no loading state in the page while this is happening, and because one of the devices was taking a very long time to respond, I was just looking at a blank Explore Libraries page with no indication what was happening.

I think this change to waiting until all the channel fetches have been resolved makes a worse user experience, because it means that seeing other libraries is limited by the loading speed of the slowest library to load.